src/TorchSharp/Utils/ShuffleGenerator.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+
+namespace TorchSharp.Utils
+{
+    public class ShuffleGenerator
+    {
+        int maxrange;
+        int prime;
+        int index;
+        int offset;
+        int runningvalue;
+
+        public ShuffleGenerator(int size)
+        {
+            var min = size / 2;
+            maxrange = size;
+            prime = selectCoPrimeResev(min, size);
+            offset = new Random().Next(size);
```

Comment:
Some of my torchvision code also relies on .NET Random() for RNG, but I'm wondering whether that is right. Maybe using Pytorch's RNG is the right thing? Not sure.

Reply:
I saw pytorch code and they are using numpy random, python random, std(in C++) random, etc. So I think .NET random will doesn't matter